src/components/RightPane/CoursePane/SearchForm/TermSelector.tsx

```diff
@@ -13,8 +13,22 @@ interface TermSelectorProps {
 }
 
 class TermSelector extends PureComponent<TermSelectorProps> {
+    
+    updateTermAndGetFormData(){
+        RightPaneStore.updateFormValue("term", RightPaneStore.getUrlTermValue()) 
+        return RightPaneStore.getFormData().term
+    };
+
+    getTerm(){
+        if(RightPaneStore.getUrlTermValue() != "null" && RightPaneStore.getUrlTermValue() != "" && RightPaneStore.getUrlTermValue() != " " ){
```

Comment:
I don't think we should invoke the function three times; just save it in a variable first and run comparisons on the calculated value. Also, if we need to ignore whitespaces, try using some variant of `String.trim` to remove the whitespaces prior to comparing.
```suggestion
       const urlTermValue = RightPaneStore.getUrlTermValue();
        if(urlTermValue != "null" && urlTermValue.trim() != "" ){
```